sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -223,6 +223,9 @@ object JdbcUtils extends Logging {
       case java.sql.Types.STRUCT        => StringType
       case java.sql.Types.TIME          => TimestampType
       case java.sql.Types.TIMESTAMP     => TimestampType
+      case java.sql.Types.TIMESTAMP_WITH_TIMEZONE
+                                        => TimestampType
+      case -101                         => TimestampType
```

Comment:
`-101` is the code Oracle databases use for `TIMESTAMP WITH TIME ZONE`. There is unfortunately no equivalent in the `java.sql.Types` Enum (https://docs.oracle.com/javase/8/docs/api/java/sql/Types.html)